src/mem/alloc.h

```diff
@@ -40,8 +40,30 @@ namespace snmalloc
     PMNotOurs = 0,
     PMSuperslab = 1,
     PMMediumslab = 2
+
+    /*
+     * Values 3 (inclusive) through SUPERSLAB_BITS (exclusive) are as yet
+     * unused.
+     *
+     * Values SUPERSLAB_BITS (inclusive) through 64 (exclusive, as it would
+     * represent the entire address space) are used for log2(size) at the
+     * heads of large allocations.  See SuperslabMap::set_large_size.
+     *
+     * Values 64 (inclusive) through 128 (exclusive) are used for entries
+     * within a large allocation.  A value of x at pagemap entry p indicates
+     * that there are at least 2^(x-64) (inclusive) and at most 2^(x+1-64)
+     * (exclusive) page map entries between p and the start of the
+     * allocation.  See SuperslabMap::set_large_size and external_address's
+     * handling of large reallocation redirections.
+     *
+     * Values 128 (inclusive) through 255 (inclusive) are as yet unused.
+     */
+
   };
 
+  /* Ensure that PageMapSuperslabKind values are actually disjoint */
+  static_assert(SUPERSLAB_BITS > 2, "Large allocations possibly too small");
```

Comment:
Why `2`?  Perhaps add a `PMEndMarker` to the enum, and then do 
```C
SUPERSLAB_BITS > next_pow2_bits_const(PMEndMarker - 1)
```

Reply:
Something seems to have gone wrong; this hunk should have been already committed as part of #100.  I must have botched rebasing; will address when I'm back at my desk.